.github/kind-config.yaml

```diff
@@ -14,6 +14,14 @@ nodes:
           taints: []
   - role: worker
     image: kindest/node:v1.19.11
+  - role: worker
+    image: kindest/node:v1.19.11
+  - role: worker
+    image: kindest/node:v1.19.11
+  - role: worker
+    image: kindest/node:v1.19.11
+  - role: worker
+    image: kindest/node:v1.19.11
```

Comment:
What's the motivation for more nodes in the kind environment?

I'm wondering overall how many of these do we need, and how they trade off the availability of resources like CPU etc. ? I presume that all of these nodes are effectively just local processes on the underlying multi-core environment, so whether we densely pack a couple of nodes or create many loosely packed nodes, the overall usage is presumably around the same. Downside with more nodes is that we would have more Cilium instances to run, sync resources between, etc., though I don't know if that's meaningful at the scale of the sorts of tests we run here.

On the other hand if for instance we think this makes the tests easier to debug when they go wrong, I could understand that rationale.

Reply:
^ Would be good to add this "why" information to the corresponding commit message too.

Reply:
Good point!
If we want to run conn tests with `--test-concurrency=3` it will spin up 3 test namespaces with 3 times more deployments. So, more resources are needed.
Maybe I'm missing something but without node increase, some conn test pods are stuck in the `Pending` status forever.

Reply:
Broadly the resources argument makes sense, but it's not obvious to me why additional nodes would make more CPU/memory available. I could also be missing something about the way that kind provisions the containers.

I think it'd be interesting to know why we hit that `Pending` status, and whether we are hitting some other known limit, whether it's configurable, or whether there's a hard requirement for more nodes. This knowledge could help inform future optimizations around how much concurrency to run, too.

Reply:
I found the root cause `:facepalm:`.
It's related to the `echoExternalNodePods` that use `Host Port`.
This commit contains the fix: https://github.com/cilium/cilium-cli/pull/2496/commits/b3f84699e268475aeb01b414c9b7e721c5c4bbe7

Thanks!